### PR TITLE
fix: Use --json flag to get runtime

### PR DESCRIPTION
**Pull request by dpgraham**
Fixes problems with https://dev.azure.com/AppiumCI/Appium%20CI/_build/results?buildId=5507

Driver Specs were not working because it wasn't parsing correctly. Tries using `--json` flag first.

### CODE REVIEW

File: lib/simctl.js [hunk 1]
```diff
@@ -621,6 +621,18 @@ async function getDevices (forSdk, platform) {
  *                  platform version.
  */
 async function getRuntimeForPlatformVersion (platformVersion, platform = 'iOS') {
+  // Try it with --json flag
+  try {
+    const {stdout} = await simExec('list', 0, ['runtimes', '--json']);
+    for (let runtime of JSON.parse(stdout).runtimes) {
+      const {version, identifier, name} = runtime;
```
Comment (mykola-mokhnach) [inline lib/simctl.js]: this could be destructed in the loop

File: test/simctl-e2e-specs.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ describe('simctl', function () {
       throw new Error('No valid SDKs');
     }
     console.log(`Found valid SDKs: ${validSdks.join(', ')}`); // eslint-disable-line no-console
-    sdk = _.last(validSdks);
+    sdk = process.env.SDK || _.last(validSdks);
```
Comment (mykola-mokhnach) [inline test/simctl-e2e-specs.js]: I would say the name of the env variable should be more specific

Comment (dpgraham) [reply test/simctl-e2e-specs.js]: Sure. I'll change it to `IOS_SDK`

**Comment by dpgraham** [comment]
Fixes pushed

File: lib/simctl.js [hunk 3]
```diff
@@ -621,6 +621,17 @@ async function getDevices (forSdk, platform) {
  *                  platform version.
  */
 async function getRuntimeForPlatformVersion (platformVersion, platform = 'iOS') {
+  // Try it with --json flag
+  try {
+    const {stdout} = await simExec('list', 0, ['runtimes', '--json']);
+    for (const {version, identifier, name} of JSON.parse(stdout).runtimes) {
+      if (version === platformVersion && name.toLowerCase().startsWith(platform.toLowerCase())) {
+        return identifier;
```
Comment (KazuCocoa) [inline lib/simctl.js]: ah...
Should here return only `12-1-1` in `com.apple.CoreSimulator.SimRuntime.iOS-12-1-1`?
(travis failure)

**Pull request merged**